src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -40,17 +43,18 @@ public class TeamExplorerContext : ITeamExplorerContext
         ILocalRepositoryModel repositoryModel;
 
         [ImportingConstructor]
-        public TeamExplorerContext(IGitHubServiceProvider serviceProvider, IVSGitExt gitExt)
-            : this(gitExt, serviceProvider)
+        TeamExplorerContext(SVsServiceProvider serviceProvider, IVSGitExt gitExt)
+            : this(gitExt, serviceProvider, ThreadHelper.JoinableTaskFactory)
         {
         }
 
-        public TeamExplorerContext(IVSGitExt gitExt, IGitHubServiceProvider serviceProvider)
+        public TeamExplorerContext(IVSGitExt gitExt, IServiceProvider serviceProvider, JoinableTaskFactory joinableTaskFactory = null)
         {
             this.gitExt = gitExt;
+            this.joinableTaskFactory = joinableTaskFactory ?? new JoinableTaskContext().Factory;
```

Comment:
What does `new JoinableTaskContext().Factory` do?

Reply:
It constructs a new task factory that consideres the current thread to be the `Main` thread.  This is useful when unit testing because it effectively turns `await.. SwitchToMainThreadAsync` into a nop.